qemu/tests/virtio_serial_file_transfer_max_ports.py

```diff
@@ -0,0 +1,47 @@
+import logging
+
+from virttest import error_context
+from virttest import env_process
+from virttest import utils_test
+from qemu.tests.virtio_serial_file_transfer import transfer_data
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test virtio serial guest file transfer with max ports.
+
+    Steps:
+    1) Boot up a VM with 30 virtserialports on one virtio-serial-pci.
+    2) Transfer data via these ports one by one.
+
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+    num_serial_ports = int(params.get("virtio_serial_ports"))
+    for i in range(2, num_serial_ports + 1):
+        params['serials'] = params.get('serials', '') + ' vs%d' % i
+        params['serial_type_vs%d' % i] = "virtserialport"
```

Comment:
To make sure the data consistency, better be:
```
serial_name = 'vs%d' % i
params['serials'] = params.get('serials', '') + serial_name
params['serial_type_%s' % serial_name] = 'virtserialport'
```

Reply:
Thanks and update.